pkg/utils/tests/tests.go

```diff
@@ -68,3 +72,34 @@ func RequireSignal(t *testing.T, ch <-chan struct{}, failMsg string) {
 		t.Fatal(failMsg)
 	}
 }
+
+// Convenience method to setup a client connection with a default timeout of 5 seconds
+func SetupClientConnWithOptsAndTimeout(t *testing.T, target string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
+	ctx, cancel := context.WithTimeout(Context(t), 5*time.Second)
+	t.Cleanup(cancel)
+
+	return grpc.DialContext(ctx, target, opts...)
+}
+
+// SetupServer is a convenience method to set up a grpc server for most testing
+// usecases.
+func SetupServer(t *testing.T, opts ...grpc.ServerOption) (net.Listener, *grpc.Server, string) {
```

Comment:
If you like this convenience method, I can refactor the other tests to consume it.